packages/apps/tools/src/components/Global/ModuleExplorer/index.tsx

```diff
@@ -60,45 +64,56 @@ const ModuleExplorer = ({
         ]}
         onReload={onReload}
         onModuleClick={({ data }) => {
-          if (isModuleLike(data)) {
-            const chainModule: IChainModule = moduleModelToChainModule(data);
-            setActiveModule(chainModule);
+          if (isCompleteModule(data)) {
+            onActiveModuleChange(data);
+            setActiveModule(data);
             setOpenedModules((prev) => {
               const alreadyOpened = prev.find((openedModule) => {
                 return (
-                  openedModule.moduleName === chainModule.moduleName &&
-                  openedModule.chainId === chainModule.chainId &&
-                  openedModule.network === chainModule.network
+                  openedModule.name === data.name &&
+                  openedModule.chainId === data.chainId &&
+                  openedModule.networkId === data.networkId
                 );
               });
 
               if (alreadyOpened) {
                 return prev;
               }
 
-              return [...prev, chainModule];
+              return [...prev, data];
             });
           }
         }}
         onExpandCollapse={onExpandCollapse}
       />
       <Editor
-        openedModules={openedModules}
-        activeModule={activeModule}
-        onActiveModuleChange={(module) => {
-          onActiveModuleChange(module);
-          setActiveModule(module);
-        }}
-        onTabClose={(module) => {
+        onChainTabClose={(module) => {
           setOpenedModules(
             openedModules.filter((openedModule) => {
               return (
-                `${openedModule.moduleName}-${openedModule.chainId}-${openedModule.network}` !==
-                `${module.moduleName}-${module.chainId}-${module.network}`
+                `${openedModule.name}-${openedModule.chainId}-${openedModule.networkId}` !==
+                `${module.name}-${module.chainId}-${module.networkId}`
               );
             }),
           );
-          // onTabClose(module);
+        }}
+        onModuleTabClose={(modules) => {
+          setOpenedModules(
+            openedModules.filter((openedModule) => {
+              return !modules.find((module) => {
+                return (
+                  `${openedModule.name}-${openedModule.chainId}-${openedModule.networkId}` ===
+                  `${module.name}-${module.chainId}-${module.networkId}`
```

Comment:
Bit of a nit but I would make a function for resolving the name, a small error could slip it and if the name ever changes, just one place to manage it.

Reply:
I've refactored it to use one function, great tip, thanks!